frame_utils.go

```diff
@@ -0,0 +1,75 @@
+package tchannel
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+type RecordingFramePool struct {
+	sync.Mutex
+
+	allocations map[*Frame]string
+	badRelease  []string
+}
+
+func NewRecordingFramePool() *RecordingFramePool {
```

Comment:
nit: suggest adding `ForTest` suffix to the method name so it's very clear this is only exported for tests (definitely unsafe for prod)